transport/split/writer.go

```diff
@@ -19,8 +19,10 @@ import (
 )
 
 type splitWriter struct {
```

Comment:
Same note as for the Dialer. Let's make this public and return the implementation in the NewWriter call, with setters for the optional parameters.